server/sonar-ce/src/test/java/org/sonar/ce/StandaloneCeDistributedInformationTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.ce;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+import org.junit.Test;
+import org.sonar.ce.taskprocessor.CeWorkerFactory;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class StandaloneCeDistributedInformationTest {
+  @Test
+  public void broadcastWorkerUUIDs_must_retrieve_from_ceworkerfactory() {
+    CeWorkerFactory ceWorkerFactory = mock(CeWorkerFactory.class);
+    StandaloneCeDistributedInformation ceCluster = new StandaloneCeDistributedInformation(ceWorkerFactory);
+
+    ceCluster.broadcastWorkerUUIDs();
+    verify(ceWorkerFactory).getWorkerUUIDs();
+  }
+
+  @Test
+  public void getWorkerUUIDs_must_be_retrieved_from_ceworkerfactory() {
+    CeWorkerFactory ceWorkerFactory = mock(CeWorkerFactory.class);
+    Set<String> workerUUIDs = ImmutableSet.of("1", "2", "3");
+    when(ceWorkerFactory.getWorkerUUIDs()).thenReturn(workerUUIDs);
+    StandaloneCeDistributedInformation ceCluster = new StandaloneCeDistributedInformation(ceWorkerFactory);
+
+    ceCluster.broadcastWorkerUUIDs();
+    assertThat(ceCluster.getWorkerUUIDs()).isEqualTo(workerUUIDs);
+  }
+
+  @Test
+  public void when_broadcastWorkerUUIDs_is_not_called_getWorkerUUIDs_is_null() {
+    CeWorkerFactory ceWorkerFactory = mock(CeWorkerFactory.class);
+    Set<String> workerUUIDs = ImmutableSet.of("1", "2", "3");
+    when(ceWorkerFactory.getWorkerUUIDs()).thenReturn(workerUUIDs);
+    StandaloneCeDistributedInformation ceCluster = new StandaloneCeDistributedInformation(ceWorkerFactory);
+
+    assertThat(ceCluster.getWorkerUUIDs()).isNull();
```

Comment:
well, this assertion is just wrong

As defined by the interface `CeDistributedInformation`, `getWorkerUUIDs` is not supposed to return `null`, ever.

I believe you have two options to fix this:

1. change `StandaloneCeDistributedInformation#getWorkerUUIDs` to return an empty collection when `broadcastWorkerUUIDs` has not yet been called. This effectively drops the contraint to call `broadcastWorkerUUIDs` prior to `getWorkerUUIDs` as everything works fine if I don't call `broadcastWorkerUUIDs`
2. throw an IllegalStateException if `broadcastWorkerUUIDs` has not been called prior to calling `getWorkerUUIDs`. This does enforces the constraint 

If your intent is really to have a calling order contraint on methods of `CeDistributedInformation`, then :

1. this should be documented in `CeDistributedInformation`
2. this should be implemented in all implementations of `CeDistributedInformation`
3. the only valid option is option 2 with the IllegalStateException

Btw, changing `CeDistributedInformation#getWorkerUUIDs`  to return `null` is not an option as it would force consuming code to check whether `getWorkerUUIDs` returns `null`.